### ISSUE TITLE
Non-contiguous subsets

### ISSUE THREAD

**Issue by cdeterman** [viennacl/viennacl-dev#228]
It would be very useful to be able to subset non-contiguous elements from matrices and vectors.  For example, taking the 1,3, and 9 elements of a vector or the 2,4, and 7 columns of a matrix.  This is currently not possible with the current `range` based methods.

I made a post on the viennacl-devel list [here](https://sourceforge.net/p/viennacl/mailman/message/34384974/) regarding this but thought best to include here as well.

**Comment by karlrupp** [comment]
You are absolutely right that this is a missing feature (general scatter/gather operations) that I want to be added rather sooner than later. The API will be most likely the following:

    viennacl::vector<double> x(N), y(N);
    viennacl::vector<int> a(N), b(N);
    ....
    x[a] = y[b];

**Comment by cdeterman** [comment]
@karlrupp Any thoughts on if this will be a possibility in the 'near' future?

**Comment by karlrupp** [comment]
@cdeterman I need this as well for optimizations in another context. I'll try to implement this next week, but I can't make promises at this point.

**Comment by cdeterman** [comment]
@karlrupp is there any way I can be of further assistance to facilitate this feature?
